basis/editors/emacs/windows/windows.factor

```diff
@@ -0,0 +1,17 @@
+! Copyright (C) 2022 nomennescio
+! See http://factorcode.org/license.txt for BSD license.
+USING: combinators.short-circuit continuations editors editors.emacs
+io.pathnames io.standard-paths kernel make math.parser
+namespaces sequences system windows.advapi32 windows.registry ;
+IN: editors.emacs.windows
+
+CONSTANT: registry-path-to-emacs "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\emacs.exe"
+
+M: windows find-emacsclient
+    {
+        [ [ HKEY_LOCAL_MACHINE registry-path-to-emacs "" query-registry
+            parent-directory "emacsclientw.exe" append-path ] [ drop f ] recover ]
```

Comment:
the whole `windows.registry` is very low level; only a thin wrapper around Win32 API calls.
I've been contemplating to build a higher level library on top of it (maybe platform independent for configurations, with backends for ".ini" files or registry?), but that seemed like overkill for this patch.

I closely modeled `query-register` like `read-registry`, which also can throw, therefore it seemed most fitting to do the recover at the use of `query-register`.

Reply:
Was the ``find-in-applications`` not working for you?  I assume this would handle some case where emacs is installed like onto your desktop or something?